bin/crew

```diff
@@ -231,14 +236,42 @@ end
 
 def generate_compatible
   @device[:compatible_packages] = []
-  Dir[CREW_PACKAGES_PATH + '*.rb'].each do |filename|
-    pkgName = File.basename filename, '.rb'
-    set_package pkgName, filename
-    if @pkg.compatibility.include? 'all' or @pkg.compatibility.include? ARCH
-      #add to compatible packages
-      @device[:compatible_packages].push(name: @pkg.name)
+  @work_q = Queue.new
+  @semaphore = Mutex.new
+  Dir[CREW_PACKAGES_PATH + '*.rb'].each{|file| @work_q << file}
+  # This creates a 10 worker thread pool
+  workers = (0...10).map do
+    Thread.new do
+    Thread.abort_on_exception = true
+      begin
+        while filename = @work_q.pop(true)
+          begin
+            pkgName = File.basename filename, '.rb'
+            puts "Checking #{pkgName} for compatibility.".orange if @opt_verbose
+            set_package pkgName, filename
+            if @pkg.compatibility.include? 'all' or @pkg.compatibility.include? ARCH
+              #add to compatible packages
+              puts "Adding #{pkgName} to compatible packages.".lightgreen if @opt_verbose
+              @semaphore.synchronize do
+                @device[:compatible_packages].push(name: @pkg.name)
+              end
+            else
+              puts "#{pkgName} is not a compatible package.".lightred if @opt_verbose
+            end
+          rescue => e
+            puts "Error with #{pkgName}.rb: #{e}".red unless e.to_s.include?('uninitialized constant') 
```

Comment:
Why is `'uninitialized constant'` not considered an error?

Reply:
That's what comes up here when there is a syntax error reported from `set_package`. This way you don't get an erroneous duplicate notification.

In reality we just need to deal with the compatibility check failing here.

Reply:
Better to move the `rescue` block to the `set_package` part?

Reply:
This entire section got refactored to remove the threading....